front/pages/w/[wId]/u/gens/index.tsx

```diff
@@ -177,10 +177,68 @@ export class FunctionSingleArgParser {
 
 export function DocumentView({
   document,
+  query,
+  owner,
 }: {
   document: GensRetrievedDocumentType;
+  query: string;
+  owner: any;
 }) {
   const provider = providerFromDocument(document);
+  const [extractedText, setExtractedText] = useState<string>("");
+
+  useEffect(() => {
+    let extractInput = [
+      {
+        query: query,
+        result: document,
```

Comment:
result is weird as an input name but that's a nit

Reply:
It's looking at a searchresult, but I don't know I guess